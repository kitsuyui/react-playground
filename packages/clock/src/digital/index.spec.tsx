import { describe, expect, it } from 'vitest'

import { render } from '@testing-library/react'

import { dateToDigital, DigitalClock } from '.'


describe('DigitalClock', () => {
  it('renders correctly', () => {
    const date = new Date('2020-01-01T12:34:56Z')
    const timezone = 'Asia/Tokyo'
    const { asFragment } = render(
      <DigitalClock date={date} timezone={timezone} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('dateToDigital', () => {
  it('converts date to digital format in Other Timezone', () => {
    const date = new Date('2020-01-01T12:34:56Z')
    const timezone = 'Asia/Manila'
    const result = dateToDigital(date, timezone)
    expect(result).toBe('08:34:56 PM (Asia/Manila)')
  })
})
