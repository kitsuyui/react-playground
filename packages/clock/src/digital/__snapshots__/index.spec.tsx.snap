// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DigitalClock > renders correctly 1`] = `
<DocumentFragment>
  <span
    style="width: 20em; font-family: monospace;"
  >
    09:34:56 PM (Asia/Tokyo)
  </span>
</DocumentFragment>
`;
